### PR TITLE
Adapt Actor Examples to Akka 2.5 #992

### PR DESCRIPTION
**Pull request by ilke-zilci**
This pr for #992:
- updates the code example for actor interaction in Java
- updates the code example for single request in actor in Java and extracts to snippet

### CODE REVIEW

**Comment by jlprat** [comment]
OK TO TEST

File: docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java [hunk 1]
```diff
@@ -170,6 +186,28 @@ public void testSingleRequestExample() {
...
+    //#single-request-in-actor-example
+      class SingleRequestInActorExample extends AbstractActor {
+      final Http http = Http.get(context().system());
+      final ExecutionContextExecutor dispatcher = context().dispatcher();
+      final Materializer materializer = ActorMaterializer.create(context());
+
+       @Override
+       public Receive createReceive() {
+         return receiveBuilder()
+                 .match(String.class, url -> pipe(fetch(url), dispatcher).to(self()))
```
Comment (jlprat) [inline docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java]: Does this line fit in the code listing on the doc page, or would make the user scroll?

File: docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java [hunk 2]
```diff
@@ -39,6 +39,22 @@
 import akka.http.javadsl.model.headers.SetCookie;
 //#collecting-headers-example
 
+
+//#single-request-in-actor-example
+import akka.actor.AbstractActor;
```
Comment (jlprat) [inline docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java]: Really cool to add the imports as well for the Java version!

Comment (ktoso) [reply docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java]: Great 👍

File: docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java (hunk 1 above)
Comment (jrudolph) [reply docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java]: Fits for me.

File: docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java [hunk 3]
```diff
@@ -170,6 +186,28 @@ public void testSingleRequestExample() {
     //#single-request-example
   }
 
+  // compile only test
+  public void singleRequestInActorExample1() {
+    //#single-request-in-actor-example
+      class SingleRequestInActorExample extends AbstractActor {
```
Comment (jrudolph) [inline docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java]: This has a bit weird indentation. If it should align properly with the imports, this line should have no indentation.

File: docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java [hunk 4]
```diff
@@ -170,6 +186,28 @@ public void testSingleRequestExample() {
...
+      final Http http = Http.get(context().system());
+      final ExecutionContextExecutor dispatcher = context().dispatcher();
+      final Materializer materializer = ActorMaterializer.create(context());
+
+       @Override
```
Comment (jrudolph) [inline docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java]: One more space in indentation than lines above ;)

File: docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java [hunk 5]
```diff
@@ -170,6 +186,28 @@ public void testSingleRequestExample() {
...
+       public Receive createReceive() {
+         return receiveBuilder()
+                 .match(String.class, url -> pipe(fetch(url), dispatcher).to(self()))
+                 .build();
+       }
+
+      CompletionStage<HttpResponse> fetch(String url) {
+        return http.singleRequest(HttpRequest.create(url));
+      }
+     }
```
Comment (jrudolph) [inline docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java]: Should align with `class` above.

File: docs/src/test/java/docs/http/javadsl/HttpServerActorInteractionExample.java [hunk 6]
```diff
@@ -121,26 +115,20 @@ static Props props() {
...
-        }).
-        match(HttpServerActorInteractionExample.GetBids.class, m -> {
+    @Override
+    public Receive createReceive() {
+      return receiveBuilder()
+        .match(
+          HttpServerActorInteractionExample.Bid.class, bid -> {
+            bids.add(bid);
+            log.info("Bid complete: {}, {}", bid.userId, bid.offer);
+            })
```
Comment (jrudolph) [inline docs/src/test/java/docs/http/javadsl/HttpServerActorInteractionExample.java]: Maybe adjust indentation here as well (though this whole file has non-standard Java indentation anyway).

File: docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java (hunk 1 above)
Comment (ilke-zilci) [reply docs/src/test/java/docs/http/javadsl/HttpClientExampleDocTest.java]: Fits for me too.

**Comment by jrudolph** [comment]
Thanks, @ktoso and @ilke-zilci!

**Comment by ilke-zilci** [comment]
Thanks for the quick reviews and fixes :) @jlprat @jrudolph @ktoso

**Pull request merged**